packages/components/atoms/f-filter-pill/src/components/FilterPill.vue

```diff
@@ -1,29 +1,183 @@
 <template>
     <div
-        :class="$style['c-filterPill']"
-        data-test-id="filter-pill-component">
-        I am a FilterPill Component (GB)
+        :class="[
+            $style['c-filterPill'], {
+                [$style['c-filterPill--selected']]: isToggleSelected,
+                [$style['c-filterPill--disabled']]: isToggleDisabled
+            }]"
+        data-test-id="filter-item">
+        <input
+            :id="`filterPillToggle-${inputId}`"
+            type="checkbox"
+            class="is-visuallyHidden"
+            :class="$style['c-filterPill-checkbox']"
+            data-test-id="filter-pill-input"
+            :tabindex="0"
+            :disabled="isToggleDisabled"
+            @change="toggleFilter">
+        <label
+            :class="$style['c-filterPill-label']"
+            :for="`filterPillToggle-${inputId}`"
+            data-test-id="filter-pill-label"
+            :tabindex="-1">
+            <tick-icon :class="$style['c-filterPill-icon']" />
+            <span
+                :class="$style['c-filterPill-text']"
+                data-test-id="filter-pill-text">
+                {{ displayText }}
+            </span>
+            <span
+                :class="$style['c-filterPill-number']"
+                data-test-id="filter-pill-number">
+                {{ displayNumber }}
+            </span>
+        </label>
     </div>
 </template>
 
 <script>
+import { TickIcon } from '@justeat/f-vue-icons';
+
 export default {
     name: 'FilterPill',
-    components: {},
+    components: {
+        TickIcon
+    },
     props: {
+        inputId: {
+            type: String,
+            default: null
+        },
+        toggleValue: {
+            type: String,
+            default: null
+        },
+        isSelected: {
+            type: Boolean,
+            default: false
+        },
+        isDisabled: {
+            type: Boolean,
+            default: false
+        },
+        displayText: {
+            type: String,
+            default: null
+        },
+        displayNumber: {
+            type: Number,
+            default: null
+        }
+    },
+    data () {
+        return {
+            isToggleSelected: false,
+            isToggleDisabled: false
+        };
+    },
+    computed: {},
+    watch: {
+        isSelected (newValue) {
+            if (newValue !== this.isToggleSelected) {
+                this.isToggleSelected = newValue;
+            }
+        },
+        isDisabled (newValue) {
+            if (newValue !== this.isToggleDisabled) {
+                this.isToggleDisabled = newValue;
+            }
+        }
+    },
+    mounted () {
```

Comment:
If you need this for SSR you might want to move the content into a different lifecycle event. I recently updated ImageTile to use created() instead of mounted(). 

Might not need updating now but something for a future ticket (potentially)

Reply:
Alix is spot on - this will likely need to go into the Created lifecycle method :)

Reply:
thanks!